src/transcoding/codec/codecs/libs/vaapi.c

```diff
@@ -238,16 +266,21 @@ tvh_codec_profile_vaapi_hevc_open(tvh_codec_profile_vaapi_t *self,
     // bit_rate or qp
     if (self->bit_rate) {
         AV_DICT_SET_BIT_RATE(opts, self->bit_rate);
+        if (self->buff_factor <= 0) {
+            self->buff_factor = 3;
+        }
         AV_DICT_SET_INT(opts, "maxrate", (self->bit_rate) * 1000, AV_DICT_DONT_OVERWRITE);
-        AV_DICT_SET_INT(opts, "bufsize", ((self->bit_rate) * 1000) * 3, AV_DICT_DONT_OVERWRITE);
+        AV_DICT_SET_INT(opts, "bufsize", ((self->bit_rate) * 1000) * self->buff_factor, AV_DICT_DONT_OVERWRITE);
         AV_DICT_SET(opts, "force_key_frames", "expr:gte(t,n_forced*3)", AV_DICT_DONT_OVERWRITE);
+        AV_DICT_SET_INT(opts, "rc_mode", self->rc_mode, AV_DICT_DONT_OVERWRITE);
+        AV_DICT_SET_INT(opts, "tier", self->tier, AV_DICT_DONT_OVERWRITE);
     }
     else {
         AV_DICT_SET_QP(opts, self->qp, 25);
     }
     // max_b_frames
     // XXX: remove when b-frames handling in vaapi_encode is fixed
-    AV_DICT_SET_INT(opts, "bf", 0, 0);
+    //AV_DICT_SET_INT(opts, "bf", 0, 0);  //seems to be fixed
```

Comment:
Please remove this completely if you are under the impression that it's no longer needed. Otherwise it'll be forgotten soon and this commented out code will stay here forever.

Reply:
I will close this request and create a new one.
In forum someone asked if it is possible to directly add -vf parameters and I already added a field to vaapi.
I need to wait until I get some response if it's working and after this I close this request and start a new one with the updated code.